packages/gatsby/src/internal-plugins/query-runner/gatsby-node.js

```diff
@@ -6,30 +6,37 @@ let components = {}
 
 const handlePageOrLayout = store => pageOrLayout => {
   // - Ensure page/layout component has a JSON file.
-  // - Add the component to Redux.
-  // - Watch the component to detect query changes..
-  const writeJsonFile = ({ jsonName }) => {
-    const dest = path.join(
-      store.getState().program.directory,
-      `.cache`,
-      `json`,
-      jsonName
-    )
-    if (!fs.existsSync(dest)) {
-      fs.writeFile(dest, `{}`, () => {})
-    }
+  const jsonDest = path.join(
```

Comment:
This should also be `joinPath`, but does it make sense to `require('../../utils/path')`? Feels odd traversing packages, but that does fix it :)